src/globus_sdk/authorizers/client_credentials.py

```diff
@@ -3,18 +3,18 @@
 import logging
 import typing as t
 
+import globus_sdk
 from globus_sdk._types import ScopeCollectionType
 from globus_sdk.scopes import scopes_to_str
 
 from .renewing import RenewingAuthorizer
 
-if t.TYPE_CHECKING:
-    from globus_sdk.services.auth import ConfidentialAppAuthClient, OAuthTokenResponse
-
 log = logging.getLogger(__name__)
 
 
-class ClientCredentialsAuthorizer(RenewingAuthorizer):
+class ClientCredentialsAuthorizer(
+    RenewingAuthorizer["globus_sdk.ClientCredentialsTokenResponse"]
```

Comment:
Why quoted?

Reply:
As a quoted forward reference, we get deferred evaluation.

We don't necessarily _have to_ defer, but given that the goal of the lazy importer is to defer the high-impact imports (namely, `requests` and `urllib3`) where possible, I think we're generally safer using deferred evaluation of annotations whenever it's an option. I've sometimes found myself disentangling the import-time behavior of modules in order to ensure that a given module doesn't eagerly import `requests` (specifically for the CLI use-case), and I'm just trying to head off this problem where possible.

Reply:
Yeah, if it were trivial to add linting, I'd do it so that we enforce "use deferred annotation syntax where possible", but it's hard to think through what such a lint would look like.

Reply:
Does this apply to return values? For example, this PR introduces several unquoted references in these same files, so is anything gained by quoting this one annotation where others are unquoted?

Reply:
Because the annotations in question are in return type annotations, `from __future__ import annotations` is enough to get deferred forward reference behavior. The special thing about the use of a generic in a class hierarchy is that it's now in a runtime context, and therefore needs to be explicitly stringified.

As another example, there's
```
if TYPE_CHECKING:
    from some_module import SomeClass

T = TypeVar("T", bound="SomeClass")  # stringified, forward reference

# or

from some_module import SomeClass

T = TypeVar("T", bound=SomeClass)  # not stringified, evaluates at runtime
```